src/plugins/intel_cpu/src/config.cpp

```diff
@@ -196,6 +196,12 @@ void Config::readProperties(const std::map<std::string, std::string> &prop) {
             else
                 IE_THROW() << "Wrong value for property key " << PluginConfigInternalParams::KEY_SNIPPETS_MODE
                             << ". Expected values: ENABLE/DISABLE/IGNORE_CALLBACK";
+        } else if (key == ov::hint::execution_mode.name()) {
+            if (val == "PERFORMANCE" || val == "UNDEFINED") {
+                executionMode = ov::hint::ExecutionMode::PERFORMANCE;
```

Comment:
```c++
core.set_property(CPU, execution_model(undefined));
auto exec_mode = core.get_property(CPU, execution_model);
```
should not `exec_mode` be equal to `UNDEFINED`?

Reply:
From my perspective, this is a tricky one.
Honestly, I am not sure why we allow to set "undefined" value at all (user could simply avoid setting the value if it is not needed, why one may need to set "undefined" explicitly?).
Then as every other parameter this one has its own default value. And what should we do if user sets it is as "undefined"?
Should we return "undefined" or the default value "performance"? Because eventually this "undefined" will be converted into "performance" anyway.

Reply:
May we need to explicitly prohibit setting `UNDEFINED` to eliminate confusion?

Reply:
As we agreed offline `UNDEFINED` value processing should be removed. Updated the PR accordingly